src/web/mjs/engine/Connector.mjs

```diff
@@ -475,7 +475,7 @@ export default class Connector {
             request = new URL(request);
         }
 
-        const requestOptions = JSON.parse(JSON.stringify(this.requestOptions));
+        const requestOptions = { ...this.requestOptions };
```

Comment:
I know this is not a good way to deep copy an object, but clonning via `JSON.parse(JSON.stringify(this.requestOptions))` loses all headers.

I checked if the `content-type` header changes in both the original and the copied object, and it turns out that only the header from the copy has changed, so I guess it's safe to use a shallow copy.
```js
const requestOptions = { ...this.requestOptions };
const graphQLRequest = new Request(request.href ? request.href : request.url, Object.assign(requestOptions, {
	method: 'POST',
	body: JSON.stringify({
		operationName,
		query,
		variables,
	})
}));
graphQLRequest.headers.set('content-type', 'application/json');
console.log(this.requestOptions.headers.get('content-type')); // null
console.log(graphQLRequest.headers.get('content-type')); // application/json
```

Reply:
JSON was used to ensure `this.requestOptions` was not accidentaly changed.
Since this is no longer the case, i propose to remove the local variable `requestOptions` completely, to avoid the ability that `this.requestOptions` can be modified through `requestOptions`.

e.g., by directly embed the suggested object spreading:
```javascript
const graphQLRequest = new Request(request.href ? request.href : request.url, {
    ...this.requestOptions,
    method: 'POST',
    body: JSON.stringify({
        operationName,
        query,
        variables,
    })
));
```
NOTE: For duplicate properties (e.g. `method`), the last assigned will be used

Reply:
Thanks for the explanations, fixed